### PR TITLE
Adding methods for default / implicit behavior in storage.

### PR DESCRIPTION
**Pull request by dhermes**
Addresses part of #337.

### CODE REVIEW

File: gcloud/storage/_implicit_environ.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module to provide implicit behavior based on enviroment.
+
+Acts as a mutable namespace to allow the datastore package to
+imply the current dataset ID and connection from the enviroment.
```
Comment (tseaver) [inline gcloud/storage/_implicit_environ.py]: Grammar tweak:  s/imply/infer/ (the "direction" is backward;  the environment implies information, but the code infers that information from the environment).

**Comment by tseaver** [comment]
LGTM aside from my note about `imply` vs `infer`.

File: gcloud/storage/_implicit_environ.py (hunk 1 above)
Comment (dhermes) [inline gcloud/storage/_implicit_environ.py]: Addressed and just rewrote into original commit.

**Comment by dhermes** [comment]
Rebased and merging.

**Pull request merged**